### ISSUE TITLE
Searching for inputs with ( but not ) causes webkit errors

### ISSUE THREAD

**Issue by mariari** [atlas-engineer/nyxt#1488]
**Describe the bug**

When one tries to call `search-buffer` and types in `(def` one will notice their terminal spews 

```
<WARN> [00:41:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: https://nyxt.atlas.engineer/start:51:25: SyntaxError: Invalid regular expression: missing )

```

This can freeze the web page if not careful.

When a `)` is given it will successfull search for a piece text, however it will highlight two positions forward as it may just ignore the ()'s

**Precise recipe to reproduce the issue**

Do the given search on the command above

**Information**
- OS name+version:
  - Arch Linux 
- Graphics card and driver:
- Desktop environment / Window manager name+version:
  - Xmonad/StumpWM  
- How you installed Nyxt (Guix pack, package manager, build from source):
  - Source  
- Information from command copy-system-information:

  If you can't run copy-system-information, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from =M-x nyxt-version= or =nyxt --version=):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**

**Comment by Ambrevar** [comment]
I looked into it, and the issue is with search-buffer.lisp code which
uses regexp to search.

I believe we should _not_ use regexps, since it's less accessible, plus
it's much slower.

@jmercouris What was the reason behind using regexp?
Can we switch to non-regexp search?

**Comment by jmercouris** [comment]
I just chose regexp because it was more powerful, we don't have to search with regexp, no.

**Comment by Ambrevar** [comment]
Wanna fix it?  You are more knowledgeable than me in this area :)

**Comment by jmercouris** [comment]
Perhaps later!

**Comment by Ambrevar** [comment]
@jmercouris Had a change to look at this?

**Comment by jmercouris** [comment]
No, not yet.

**Comment by Ambrevar** [comment]
I got down to it, fixed with 78fedea7efdde22c8c6be1ffa3e95b0ecdad2049.
